lib/cosmos/interfaces/protocols/stream_protocol.rb

```diff
@@ -38,50 +36,24 @@ class StreamProtocol
     #   that will be searched for in the raw {Stream}. Bytes encountered before
     #   this pattern is found are discarded.
     # @param fill_fields [Boolean] Fill any required fields when writing packets
-    def initialize(discard_leading_bytes = 0, sync_pattern = nil, fill_fields = false)
+    def configure_stream_protocol(discard_leading_bytes = 0, sync_pattern = nil, fill_fields = false)
```

Comment:
Since this is a module now and not a class I couldn't do things in initialize. Thus I added this configure_stream_protocol method to set things up. I don't really like that but I'm not sure how else to configure the instance variables and setup the module.